source/components/defining-a-component.md

```diff
@@ -81,3 +81,57 @@ example, if you have a component called `blog-post`, you would create a
 file at `app/components/blog-post.js`. If your component was called
 `audio-player-controls`, the file name would be at
 `app/components/audio-player-controls.js`.
+
+### Dynamically rendering a component
+
+Sometimes, there is the need to decide which component to render at run time,
+instead of always rendering the same component with the usual `{{my-component}}`
+syntax.
+
+For this purpose, you can use the `{{component}}` helper. The first parameter
+of the helper is the name of a component to render, as a string. So if you have `{{component 'blog-post'}}`, that is just the same as just `{{blog-post}}`.
+
+The real value of `{{component}}` comes from being able to dynamically pick
+the component being rendered:
+
+```app/templates/index.hbs
```

Comment:
This example is quite long without a lot of context. Maybe some in-line comments or narrative interspersed to explain to new users what's going on?